tests/cephfs/cephfs_nfs/test_cephfs_nfs_ha_snapshot_access.py

```diff
@@ -0,0 +1,338 @@
+        log.info("Cleaning Up")
+        log.info("Remove snapshots")
+        for subvol in subvol_list:
+            if snap_list:
+                for snap_name in snap_list:
+                    snapshot = {
+                        "vol_name": fs_name,
+                        "subvol_name": subvol,
+                        "snap_name": snap_name,
+                        "group_name": "subvolume_group1",
+                    }
+                    fs_util.remove_snapshot(client1, **snapshot)
+
+        log.info("Unmount NFS export")
+        for mount in nfs_mount_list:
+            client1.exec_command(sudo=True, cmd=f"umount -l {mount}", check_ec=False)
+        log.info("Removing the Export")
+        for nfs_export_name in export_name_list:
+            client1.exec_command(
+                sudo=True,
+                cmd=f"ceph nfs export delete {nfs_name} {nfs_export_name}",
+                check_ec=False,
+            )
+        log.info("Removing NFS Cluster")
+        client1.exec_command(
+            sudo=True,
+            cmd=f"ceph nfs cluster rm {nfs_name}",
+            check_ec=False,
+        )
+        log.info("Removing Subvolumes and Subvolumegroup")
+        for subvol in subvol_list:
+            subvolume = {
+                "vol_name": fs_name,
+                "subvol_name": subvol,
+                "group_name": "subvolume_group1",
+            }
+            fs_util.remove_subvolume(client1, **subvolume)
+        fs_util.remove_subvolumegroup(client1, **subvolumegroup)
+
+
+def validate_snap_nfs(client, mount_list, snap_list):
+    for mount in mount_list:
+        cmd_out, cmd_rc = client.exec_command(sudo=True, cmd=f"ls {mount}/.snap")
+        log.info(cmd_out)
+        for snap_name in snap_list:
+            if snap_name not in cmd_out:
+                cmd_list = [
+                    "ceph orch ps --daemon_type=nfs",
+                    "ceph nfs cluster info cephfs-nfs",
+                ]
+                for cmd in cmd_list:
+                    client.exec_command(sudo=True, cmd=cmd)
+                log.error(f"Failed to view snapshot {snap_name} in NFS mount {mount}")
+
+
+def create_snapshot_wrapper(client1, fs_util, subvol_list, snap_list):
+    with parallel() as p:
+        for subvol in subvol_list:
+            p.spawn(create_snapshot, client1, fs_util, subvol, snap_list)
+
+
+def create_snapshot(client1, fs_util, subvol, snap_list):
+    for snap_name in snap_list:
+        snapshot = {
+            "vol_name": "cephfs",
+            "subvol_name": subvol,
+            "snap_name": snap_name,
+            "group_name": "subvolume_group1",
+        }
+        fs_util.create_snapshot(client1, **snapshot)
+        time.sleep(20)
+
+
+def write_io_wrapper(client1, nfs_mount_list, dir_name, run_time):
+    with parallel() as p:
+        for mount in nfs_mount_list:
+            p.spawn(write_io, client1, mount, dir_name, run_time)
+
+
+def write_io(client1, mount_dir, dir_name, run_time):
```

Comment:
Hi Suma,
This function is redundant I guess. Can you use the function in cephfs_utilsV1.py

Reply:
They are not reduntant amar, am using /dev/random and bs=10M not 100M, am explicitly creating io file. I prefer to have this io pattern for this automation, hence a local io module.

Reply:
@sumabai Any reason we are specifically using the IO pattern mentioned in this test? The method write_io() in cephfs_utilsV1 can be modified to write using /dev/random instead of /dev/zero to write IO in all cases. 
@hkadam134 @AmarnatReddy Thoughts on write_io() in cephfs_utilsV1 ?